synfig-core/src/synfig/rendering/surface.h

```diff
@@ -378,6 +371,14 @@ class SurfaceResource: public etl::shared_object
 			outTokens.push_back(i->first);
 		return !surfaces.empty();
 	}
+	Surface::Handle get_surface(
```

Comment:
It seems alright, but I think there's a reason for not being public available.
I believe you should do like I saw now in centerlinepolygonizer.cpp :
`  rendering::SurfaceResource::LockRead<rendering::SurfaceSW> lock( ras->rendering_surface );`
But instead of `rendering::SurfaceSW` change it to `rendering::Surface` directly (it's safer)